File: 022-栈的压入弹出序列/022-IsPopOrder.py
# -*- coding:utf-8 -*-
class Solution:
    def IsPopOrder(self, pushV, popV):
        # write code here
        if len(pushV)!=len(popV):
            return False
        stack=[]
        idx=0
        for v in pushV:
            stack.append(v)
            while stack and stack[-1]==popV[idx]:
                idx+=1
                stack.pop()
        return len(stack)==0

File: 022-栈的压入弹出序列/022-IsPopOrder.cpp
class Solution {
public:
    bool IsPopOrder(vector<int> pushV,vector<int> popV) {
        if(pushV.size()!=popV.size()){
            return false;
        }
        stack<int> f;
        int idx=0;
        for(auto v:pushV){
            f.push(v);
            while(!f.empty() && f.top()==popV[idx]){
                f.pop();
                idx++;
            }
        }
        return f.empty();
    }
};
